pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -460,6 +460,16 @@ protected void internalCreateNonPartitionedTopic(boolean authoritative) {
         if (partitionMetadata.partitions > 0) {
             log.warn("[{}] Partitioned topic with the same name already exists {}", clientAppId(), topicName);
             throw new RestException(Status.CONFLICT, "This topic already exists");
+        } else {
+            if (topicName.isPartitioned()) {
+                final TopicName partitionedTopicName = TopicName.get(topicName.getPartitionedTopicName());
+                partitionMetadata = fetchPartitionedTopicMetadata(pulsar(), partitionedTopicName);
```

Comment:
I think we shouldn't allow creating a topic with the partitioned topic name style even if the partition is not existing. That would be weird there has a non-partitioned topic with a partitioned topic name when you listing topics.  We can improve this check in the `validateNonPartitionTopicName`. What do you think?

Reply:
Sounds good. I'll use the suggested way to fix.

Reply:
After rethinking for a while, I think a better way is to check if `/managed-ledgers/<tenant>/<namespace>/<topic>` exists.

No matter if disallow creating `xxx-partition-i` or the way I adopted, the compatibility will be broken. Because if users upgrade their pulsar from a low version, they may rely on creating a `xxx-partition-0` to create missing partitions instead of the REST API, as well as some old code of other clients.